Mark fully processed normal records in the pipeline

mir-am opened the pull request:
As we discussed, in the normal, we should mark fully processed normal records in the `ingest_artifacts` table in the DB. This way, we can avoid processing them in the priority pipeline.
This PR adds a tiny plugin that performs the above-described functionality. The plugin consumes normal records from the callable-indexer's `.out`, which is the last plug-in in the pipeline.

proksch commented:
As discussed offline, the missing piece is probably a missing dependency to the new plugin in the loader `pom.xml`.

mir-am commented:
> As discussed offline, the missing piece is probably a missing dependency to the new plugin in the loader `pom.xml`.

I've addressed your comments and the plug-in is also tested in the DC and it works as expected:
- Marks fully processed records in the normal lane in the `ingested_artifacts` table as `PRIORITY`. This prevents these records from being processed again in the priority lane.

proksch reviewed (approved):
Did some minor refactoring, but generally LGTM!
